samcli/commands/local/lib/local_lambda.py

```diff
@@ -243,33 +247,37 @@ def _make_env_vars(self, function: Function) -> EnvironmentVariables:
         else:
             LOG.debug("No environment variables found for function '%s'", name)
 
-        # This could either be in standard format, or a CloudFormation parameter file format.
+        # This could either be in standard format, or a CloudFormation parameter file format, or mix of both.
         #
         # Standard format is {FunctionName: {key:value}, FunctionName: {key:value}}
         # CloudFormation parameter file is {"Parameters": {key:value}}
+        # Mixed format is {FunctionName: {key:value}, "Parameters": {key:value}}
```

Comment:
What happens if a function name is `Parameters`?

Reply:
I had the same question, but it seems it is how we currently handle global parameters. If we changed it, we may break our customers

Reply:
Yes I agree with @moelasmar. I checked and sam build handles the global env variables in a similar way [here](https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/build/app_builder.py#L934-L948).  If a function name was Parameters then with the current code it would not allow reading specific env variables for other lambda functions. With the new changes if function specific env vars are given, they would have precedence over the "Parameters" env vars which is similar to sam build.